### PR TITLE
Added async input/ouput to the server

### PR DESCRIPTION
**Pull request by AxlCruzTKH**
Not sure what happened to the old branch during rebasing.

### CODE REVIEW

File: aepsych/server/sockets.py [hunk 1]
```diff
@@ -7,14 +7,17 @@
...
 import socket
 import sys
+import time
 
 import aepsych.utils_logging as utils_logging
 import numpy as np
 import zmq
 
 logger = utils_logging.getLogger(logging.INFO)
+BAD_REQUEST = "bad request"
```
Comment (crasanders) [inline aepsych/server/sockets.py:20]: We shouldn't define the same constant in multiple places. Define it in one place and import it wherever you need it.

File: tests/test_server.py [hunk 2]
```diff
@@ -560,15 +534,77 @@ def test_pandadf_dump_flat(self):
...
+        message3 = {"message": {"target": "test request"}}  # valid message
+        message_list = [message1, message2, json.dumps(message3)]
+
+        self.s.socket.conn = MagicMock()
+
+        for i, message in enumerate(message_list):
+            select.select = MagicMock(return_value=[[self.s.socket.conn], [], []])
+            self.s.socket.conn.recv = MagicMock(return_value=message)
+            if i != 2:
+                self.assertEqual(self.s.socket.receive(False), "bad request")
```
Comment (crasanders) [inline tests/test_server.py:555]: Make use of the BAD_REQUEST constant.

**Review by crasanders — CHANGES_REQUESTED** [review]
This is close. See my comments about the constant.

File: aepsych/server/sockets.py (hunk 1 above)
Comment (AxlCruzTKH) [reply aepsych/server/sockets.py:20]: Done. Defined in sockets, imported everywhere else.

File: tests/test_server.py (hunk 2 above)
Comment (AxlCruzTKH) [reply tests/test_server.py:555]: Change made.

**Comment by crasanders** [comment]
@AxlCruzTKH, I rebased this and tried running the tests, and ran into some issues in test_server.py. Do you want to look into this?

**Pull request closed without merging**